Updated important upgrade note (#6303)
mattermost/docs: source/install/self-managed-changelog.md

### Important Upgrade Notes

- Added a new index on ``Posts(OriginalId)``. For a database with 11.8 million posts, on a machine with a i7-11800H CPU (8 cores, 16 threads), 32GiB of RAM and SSD, the index creation takes 98.51s on MYSQL and 2.6s on PostgreSQL. 
- In PostgreSQL databases, the ``Posts`` table will be locked during index creation. To avoid locking this table, admins can create the index manually before performing the upgrade using the following non-blocking query: ``CREATE INDEX CONCURRENTLY idx_posts_original_id ON Posts(OriginalId);``. 

> - Admins managing PostgreSQL deployments containing fewer posts may prefer that the migration process create the index, and accept that ``Posts`` table will remain locked until the migration is complete.

Reviewer: I'm very scared of suggesting a grammar fix to a technical writer, but here I go. I'll be happy to stand corrected and learn some new grammar if this is wrong.
```suggestion
- Admins managing PostgreSQL deployments containing fewer posts may prefer that the migration process creates the index, and accept that ``Posts`` table will remain locked until the migration is complete.
```

> - Admins managing servers and PostgreSQL databases containing fewer posts may prefer that the migration process create the index, and accept that ``Posts`` table will remain locked until the migration is complete.

Reviewer: Ah, the list is even more clear, thanks! And sorry for the back and forth on this, but what about something like this to simplify?
```suggestion
- Admins managing PostgreSQL deployments containing fewer posts may prefer that the migration process create the index, and accept that ``Posts`` table will remain locked until the migration is complete.
```

### Important Upgrade Notes

> - Added a new index on ``Posts(OriginalId)``. For a database with 11.8 million posts, on a machine with a i7-11800H CPU (8 cores, 16 threads), 32GiB of RAM and SSD, the index creation takes 98.51s on MYSQL and 2.6s on PostgreSQL. Database tables will be locked during index creation. To avoid locking the database tables, admins can create the index manually before performing the upgrade using the following non-blocking query: ``CREATE INDEX CONCURRENTLY idx_posts_original_id ON Posts(OriginalId);``. Admins managing servers with fewer posts may prefer that the migration process create the index, and accept that tables will be locked until the migration is complete.

Author: ```suggestion
- Added a new index on ``Posts(OriginalId)``. For a database with 11.8 million posts, on a machine with a i7-11800H CPU (8 cores, 16 threads), 32GiB of RAM and SSD, the index creation takes 98.51s on MYSQL and 2.6s on PostgreSQL. The ``Posts`` database table will be locked during index creation. To avoid locking the ``Posts`` table in PostgreSQL databases, admins can create the index manually before performing the upgrade using the following non-blocking query: ``CREATE INDEX CONCURRENTLY idx_posts_original_id ON Posts(OriginalId);``. Admins managing servers with fewer posts may prefer that the migration process create the index, and accept that ``Post`` table will remain locked until the migration is complete.
```

> - Added a new index on ``Posts(OriginalId)``. For a database with 11.8 million posts, on a machine with a i7-11800H CPU (8 cores, 16 threads), 32GiB of RAM and SSD, the index creation takes 98.51s on MYSQL and 2.6s on PostgreSQL. The ``Posts`` database table will be locked during index creation. To avoid locking the ``Posts`` table in PostgreSQL databases, admins can create the index manually before performing the upgrade using the following non-blocking query: ``CREATE INDEX CONCURRENTLY idx_posts_original_id ON Posts(OriginalId);``. Admins managing servers with fewer posts may prefer that the migration process create the index, and accept that ``Post`` table will remain locked until the migration is complete.

Author: ```suggestion
- Added a new index on ``Posts(OriginalId)``. For a database with 11.8 million posts, on a machine with a i7-11800H CPU (8 cores, 16 threads), 32GiB of RAM and SSD, the index creation takes 98.51s on MYSQL and 2.6s on PostgreSQL. The ``Posts`` database table will be locked during index creation. To avoid locking the ``Posts`` table in PostgreSQL databases, admins can create the index manually before performing the upgrade using the following non-blocking query: ``CREATE INDEX CONCURRENTLY idx_posts_original_id ON Posts(OriginalId);``. Admins managing servers with fewer posts may prefer that the migration process create the index, and accept that ``Posts`` table will remain locked until the migration is complete.
```

> - Added a new index on ``Posts(OriginalId)``. For a database with 11.8 million posts, on a machine with a i7-11800H CPU (8 cores, 16 threads), 32GiB of RAM and SSD, the index creation takes 98.51s on MYSQL and 2.6s on PostgreSQL. The ``Posts`` database table will be locked during index creation. To avoid locking the ``Posts`` table in PostgreSQL databases, admins can create the index manually before performing the upgrade using the following non-blocking query: ``CREATE INDEX CONCURRENTLY idx_posts_original_id ON Posts(OriginalId);``. Admins managing servers with fewer posts may prefer that the migration process create the index, and accept that ``Posts`` table will remain locked until the migration is complete.

Reviewer: This part also applies only to Postgres:
> The ``Posts`` database table will be locked during index creation

> - Added a new index on ``Posts(OriginalId)``. For a database with 11.8 million posts, on a machine with a i7-11800H CPU (8 cores, 16 threads), 32GiB of RAM and SSD, the index creation takes 98.51s on MYSQL and 2.6s on PostgreSQL. The ``Posts`` database table will be locked during index creation. To avoid locking the ``Posts`` table in PostgreSQL databases, admins can create the index manually before performing the upgrade using the following non-blocking query: ``CREATE INDEX CONCURRENTLY idx_posts_original_id ON Posts(OriginalId);``. Admins managing servers with fewer posts may prefer that the migration process create the index, and accept that ``Posts`` table will remain locked until the migration is complete.

Author: ```suggestion
- Added a new index on ``Posts(OriginalId)``. For a database with 11.8 million posts, on a machine with a i7-11800H CPU (8 cores, 16 threads), 32GiB of RAM and SSD, the index creation takes 98.51s on MYSQL and 2.6s on PostgreSQL. 
- In PostgreSQL databases, the ``Posts`` table will be locked during index creation. To avoid locking this table, admins can create the index manually before performing the upgrade using the following non-blocking query: ``CREATE INDEX CONCURRENTLY idx_posts_original_id ON Posts(OriginalId);``. 
- Admins managing servers and PostgreSQL databases containing fewer posts may prefer that the migration process create the index, and accept that ``Posts`` table will remain locked until the migration is complete.
```
